cmd/gossamer/configcmd.go

```diff
@@ -193,31 +194,49 @@ func loadStateAndRuntime(ss *state.StorageState, ks *keystore.Keystore) (*runtim
 	return runtime.NewRuntime(code, ss, ks)
 }
 
-// getConfig checks for config.toml if --config flag is specified and sets CLI flags
-func getConfig(ctx *cli.Context) (*dot.Config, error) {
-	currentConfig := gssmr.DefaultConfig()
+// getConfig gets the configuration for the node using --node and/or --config,
+// then applies the remaining cli flag options to the configuration
+func getConfig(ctx *cli.Context) (cfg *dot.Config, err error) {
+
+	// check --node flag and apply node defaults to config
+	if name := ctx.GlobalString(NodeFlag.Name); name != "" {
+		switch name {
+		case "gssmr":
+			log.Trace("[gossamer] Using node implementation", "name", name)
+			cfg = gssmr.DefaultConfig()
+		case "ksmcc":
+			log.Trace("[gossamer] Using node implementation", "name", name)
+			cfg = ksmcc.DefaultConfig()
+		}
+	} else {
+		log.Trace("[gossamer] Using node implementation", "name", "gssmr")
```

Comment:
would be nice to let the user know their input was invalid

Reply:
it's not invalid, this is just the default if no --node flag option is provided

Reply:
but I need to add a `default` to the switch to let the user know their input was invalid